system/logaccess/Grafana/CurlClient/GrafanaCurlClient.cpp

```diff
@@ -0,0 +1,824 @@
+        else if (streq(logMapType, "workunits"))
+        {
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_workunitsColumn = logMap.queryProp(logMapSearchColAtt);
+        }
+        else if (streq(logMapType, "components"))
+        {
+            if (logMap.hasProp(logMapIndexPatternAtt))
+                if (strcmp(logMap.queryProp(logMapIndexPatternAtt), "stream")==0)
+                    m_componentsColumn.isStream = true;
+
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_componentsColumn.name = logMap.queryProp(logMapSearchColAtt);
+        }
+        else if (streq(logMapType, "class"))
+        {
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_classColumn = logMap.queryProp(logMapSearchColAtt);
+        }
+        else if (streq(logMapType, "audience"))
+        {
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_audienceColumn = logMap.queryProp(logMapSearchColAtt);
+        }
+        else if (streq(logMapType, "instance"))
+        {
+            if (logMap.hasProp(logMapIndexPatternAtt))
+                if (strcmp(logMap.queryProp(logMapIndexPatternAtt), "stream")==0)
+                    m_instanceColumn.isStream = true;
+
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_instanceColumn.name = logMap.queryProp(logMapSearchColAtt);
+        }
+        else if (streq(logMapType, "node"))
+        {
+            if (logMap.hasProp(logMapIndexPatternAtt))
+                if (strcmp(logMap.queryProp(logMapIndexPatternAtt), "stream")==0)
+                    m_nodeColumn.isStream = true;
+
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_nodeColumn.name = logMap.queryProp(logMapSearchColAtt);
+        }
+        else if (streq(logMapType, "host"))
+        {
+            OWARNLOG("%s: 'host' LogMap entry is NOT supported!", COMPONENT_NAME);
+        }
+        else if (streq(logMapType, "pod"))
+        {
+            if (logMap.hasProp(logMapIndexPatternAtt))
+                if (strcmp(logMap.queryProp(logMapIndexPatternAtt), "stream")==0)
+                    m_podColumn.isStream = true;
+
+            if (logMap.hasProp(logMapSearchColAtt))
+                m_podColumn.name = logMap.queryProp(logMapSearchColAtt);
+        }
+        else
+        {
+            ERRLOG("Encountered invalid LogAccess field map type: '%s'", logMapType);
+        }
+    }
+
+    DBGLOG("%s: targeting: '%s' - datasource: '%s'", COMPONENT_NAME, m_grafanaConnectionStr.str(), m_dataSourcesAPIURI.str());
+}
+
+IRemoteLogAccessStream * GrafanaLogAccessCurlClient::getLogReader(const LogAccessConditions & options, LogAccessLogFormat format)
+{
+    return nullptr;
+//    return getLogReader(options, format, defaultMaxRecordsPerFetch);
+}
+
+IRemoteLogAccessStream * GrafanaLogAccessCurlClient::getLogReader(const LogAccessConditions & options, LogAccessLogFormat format, unsigned int pageSize)
+{
+    return nullptr;
```

Comment:
Are these functions planning on being implemented (also line 812)

Reply:
yes, I have them implemented, and should have pushed last night after testing. Will include as part of this code review cycle